src/sentry/integrations/slack/tasks.py

```diff
@@ -170,6 +180,30 @@ def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=No
             redis_rule_status.set_value("failed")
             return
 
+    try:
+        mapped_ids = get_slack_channel_ids(organization, user, data)
+    except (serializers.ValidationError, ChannelLookupTimeoutError, InvalidTriggerActionError) as e:
+        # channel doesn't exist error or validation error
+        logger.info(
+            "get_slack_channel_ids.failed",
+            extra={
+                "exception": e,
+            },
+        )
+        redis_rule_status.set_value("failed")
+        return
+
+    for trigger in data["triggers"]:
+        for action in trigger["actions"]:
+            if action["type"] == "slack":
+                if action["targetIdentifier"] in mapped_ids:
+                    action["input_channel_id"] = mapped_ids[action["targetIdentifier"]]
+                    # We can early exit because we couldn't map this action's slack channel name to a slack id
+                    # This is a fail safe, but I think we shouldn't really hit this.
+                else:
```

Comment:
Crap the comment belongs below the else but I'm not triggering CI again just for this. I'll fix it in another PR haha.